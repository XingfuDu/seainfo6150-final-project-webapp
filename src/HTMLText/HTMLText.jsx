import React from "react";
import PropTypes from "prop-types";
import classes from "./HTMLText.module.css";

const HTMLText = (props) => {
  return <div className={classes.description} dangerouslySetInnerHTML={{ __html: props.text }} />;
};

HTMLText.propTypes = {
    text: PropTypes.string.isRequired
}
export default HTMLText;
