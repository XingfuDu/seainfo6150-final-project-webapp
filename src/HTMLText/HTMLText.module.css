.description {
    color: #3d5a80;
    border-top: 1px white solid;
    padding-top: 30px;
}