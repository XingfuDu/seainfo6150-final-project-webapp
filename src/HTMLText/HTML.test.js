import React from 'react';
import HTMLText from './HTMLText';
import { render } from "@testing-library/react";

describe('HTMLText tests', () => {
	it('renders correctly', () => {
		const { container } = render(<HTMLText text="text"/>);
		expect(container).toMatchSnapshot();
	});
});
