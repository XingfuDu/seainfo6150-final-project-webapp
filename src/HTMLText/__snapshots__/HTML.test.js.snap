// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTMLText tests renders correctly 1`] = `
<div>
  <div
    class="description"
  >
    text
  </div>
</div>
`;
